pkg/api/validation/validation.go

```diff
@@ -440,9 +440,26 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource) errs.Valid
 
 func validateGitRepoVolumeSource(gitRepo *api.GitRepoVolumeSource) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
+	target := gitRepo.Directory
+
 	if gitRepo.Repository == "" {
 		allErrs = append(allErrs, errs.NewFieldRequired("repository"))
 	}
+	if path.IsAbs(target) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("directory", target, "must not be an absolute path"))
+	}
+	items := strings.Split(target, string(os.PathSeparator))
+	for _, item := range items {
+		if item == ".." {
+			allErrs = append(allErrs, errs.NewFieldInvalid("directory", target, "must not contain \"..\""))
+		}
+		if item == " " {
```

Comment:
what about trim and a check for empty string?  that would find N spaces, not just 1.